### PR TITLE
Set up user model and authentication

### LINKED ISSUE #17: Set up authentication and user model
Includes:
- [x] Install devise and devise-remote-user
- [x] Configure devise-remote-user
- [x] Generate user model via devise
- [x] Add Shibboleth middleware support including testing hooks
- [x] Create login controller per other Shibbolized projects

A possible pattern in H2: https://github.com/sul-dlss/happy-heron/commit/7546340c242df129f2e1701c25ce83514fe0456e

### PR DESCRIPTION
**Pull request by mjgiarlo**
# Why was this change made? 🤔

Fixes #17

Includes:
* Install and setup `devise` gem
* Use devise-remote-user for Shibboleth authentication
* Use FactoryBot to handle fixture-like data

# How was this change tested? 🤨

CI + local testing

### CODE REVIEW

File: app/controllers/stats_controller.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
-# frozen_string_literal: true
-
```
Comment (mjgiarlo) [inline app/controllers/stats_controller.rb:2]: We are now free from caring about this.

File: .gitignore [hunk 2]
```diff
@@ -42,3 +42,6 @@
 config/settings.local.yml
 config/settings/*.local.yml
 config/environments/*.local.yml
+
+# RSpec examples
+spec/examples.txt
```
Comment (justinlittman) [inline .gitignore:47]: We usually change this default to '.rspec_status', I believe.

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -50,49 +50,47 @@
...
+  # you care about by tagging them with `:focus` metadata. When nothing
+  # is tagged with `:focus`, all examples get run. RSpec also provides
+  # aliases for `it`, `describe`, and `context` that include `:focus`
+  # metadata: `fit`, `fdescribe` and `fcontext`, respectively.
+  config.filter_run_when_matching :focus
+
+  # Allows RSpec to persist some state between runs in order to support
+  # the `--only-failures` and `--next-failure` CLI options. We recommend
+  # you configure your source control system to ignore this file.
+  config.example_status_persistence_file_path = "spec/examples.txt"
```
Comment (justinlittman) [inline spec/spec_helper.rb:63]: We usually change this default to `'.rspec_status'`, I believe.

**Review by justinlittman — APPROVED** [review]
Approve pending consideration of comment.

**Pull request merged**